# Copyright © Sébastien Luttringer

# This patch is free software: you can redistribute it and/or modify
# it under the terms of the GNU Lesser General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# This patch is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this patch. If not, see <http://www.gnu.org/licenses/>.

# perl 5.22 raise errors when defined is used on arrays

--- a/lib/BackupPC/CGI/Browse.pm	2015-06-01 09:05:54.000000000 +0200
+++ b/lib/BackupPC/CGI/Browse.pm	2015-07-01 14:34:28.480199121 +0200
@@ -65,7 +65,7 @@
     #
     # default to the newest backup
     #
-    if ( !defined($In{num}) && defined(@Backups) && @Backups > 0 ) {
+    if ( !defined($In{num}) && @Backups && @Backups > 0 ) {
         $i = @Backups - 1;
         $num = $Backups[$i]{num};
     }
